Code-Code/TypePrediction-TypeScript/README.md

```diff
@@ -0,0 +1,121 @@
+# CodeXGLUE -- Type Prediction -- TypeScript
+
+## Task Definition
+Given a sequence of source code, the task is to predict the correct type for a particular variable, parameter, or function. Type prediction is an important task for software developement, especially in dynamically-typed environments, which can benefit from stronger type checking while maintaining the advantages of dynamic typing. We treat the task as sequence tagging task, similar to Named-Entity Recognition (NER) in NLP.
+
+
+### Dataset
+
+The dataset originates from the MSR '22 paper [ManyTypes4TypeScript: A Comprehensive TypeScript Dataset for Sequence-Based Type Inference](). The dataset is available for download on Zenodo. 
+
+### Download and Preprocess
+
+1.Download dataset either using wget per below or by clicking on the [link](https://zenodo.org/record/6387001/files/ManyTypes4TypeScript.tar.gz). Unzip and run the preprocess script with chosen parameters like type vocabulary size. To use the default type vocabulary size of 50k types, train#.jsonl, test.jsonl, valid.jsonl are already available in the main directory.
+
+```shell
+cd CodeXGLUE/Code-Code/TypePrediction-TypeScript/dataset
+wget https://zenodo.org/record/6336114/files/ManyTypes4TypeScript.tar.gz -O ManyTypes4TypeScript.tar.gz
+tar -xvzf ManyTypes4TypeScript.tar.gz
+
+```
+
+2.Preprocess dataset
+Use the process_datafiles.py in dataset folder.
+
+```shell
+python3 process_datafiles.py <vocab-size> 
```

Comment:
I think it should be `python3 process_datafiles.py -v <vocab-size>` based on the codes.

Reply:
You are correct. Will resolve in new commit.